groups/bsl/bslstl/bslstl_optional.h

```diff
@@ -277,152 +267,100 @@ struct Optional_RemoveCVRef {
         typename bsl::remove_reference<TYPE>::type>::type type;
 };
 
-
-template <class TYPE, class OPT_TYPE>
-struct Optional_ConvertsFrom
-: bsl::integral_constant<
-      bool,
-      bsl::is_convertible<const OPT_TYPE&, TYPE>::value ||
-          bsl::is_convertible<OPT_TYPE&, TYPE>::value ||
-          bsl::is_convertible<const OPT_TYPE, TYPE>::value ||
-          bsl::is_convertible<OPT_TYPE, TYPE>::value ||
-          Optional_IsConstructible<TYPE, const OPT_TYPE&, false>::value ||
-          Optional_IsConstructible<TYPE, OPT_TYPE&, false>::value ||
-          Optional_IsConstructible<TYPE, const OPT_TYPE, false>::value ||
-          Optional_IsConstructible<TYPE, OPT_TYPE, false>::value> {
-    // This metafunction is derived from 'bsl::true_type' if 'TYPE' can be
-    // converted from an 'OPT_TYPE', and from 'bsl::false_type' otherwise.
-};
+// As in 'std' implementation, if the 'TYPE' converts from any value category
+// of an optional type 'OPT_TYPE', we consider it convertible from that
+// optional type.
+#define BSLSTL_OPTIONAL_CONVERTS_FROM(TYPE, OPT_TYPE)                         \
+    (bsl::is_convertible<const OPT_TYPE&, TYPE>::value ||                     \
+     bsl::is_convertible<OPT_TYPE&, TYPE>::value ||                           \
+     bsl::is_convertible<const OPT_TYPE, TYPE>::value ||                      \
+     bsl::is_convertible<OPT_TYPE, TYPE>::value ||                            \
+     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, const OPT_TYPE&, false)::value || \
+     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, OPT_TYPE&, false)::value ||       \
+     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, const OPT_TYPE, false)::value ||  \
+     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, OPT_TYPE, false)::value)
```

Comment:
```suggestion
     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, const OPT_TYPE&, false) ||        \
     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, OPT_TYPE&, false) ||              \
     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, const OPT_TYPE, false) ||         \
     BSLSTL_OPTIONAL_IS_CONSTRUCTIBLE(TYPE, OPT_TYPE, false))
```

Reply:
I think having a macro followed by `::value` is odd and isn't consistent with other macros.  Please apply this change in the remainder of the file.